File: src/dataset/typescript/run-length-encoding_Exercism-typescript-shybyte/gemini/GeminiFlash_run-length-encoding_v3.ts
export function encode(input: string) {
    let encoded = "";
    let count = 1;
    for (let i = 0; i < input.length; i++) {
        if (i + 1 < input.length && input[i] === input[i + 1]) {
            count++;
        } else {
            encoded += (count > 1 ? count : "") + input[i];
            count = 1;
        }
    }
    return encoded;
}

export function decode(input: string) {
    let decoded = "";
    let countStr = "";
    for (let i = 0; i < input.length; i++) {
        if (/\d/.test(input[i])) {
            countStr += input[i];
        } else {
            const count = countStr ? parseInt(countStr, 10) : 1;
            decoded += input[i].repeat(count);
            countStr = "";
        }
    }
    return decoded;
}

export default { encode, decode };

File: src/dataset/go/run_length_encoding_Exercism_go_ThomasZumsteg/gemini/GeminiFlash_run_length_encoding_v3.go
package encode

import (
	"strconv"
	"strings"
)

func RunLengthEncode(input string) string {
	if input == "" {
		return ""
	}

	var sb strings.Builder
	sb.Grow(len(input)) // Pre-allocate memory

	last := rune(input[0])
	count := 1

	for i := 1; i < len(input); i++ {
		char := rune(input[i])
		if char == last {
			count++
		} else {
			if count > 1 {
				sb.WriteString(strconv.Itoa(count))
			}
			sb.WriteRune(last)
			last = char
			count = 1
		}
	}

	if count > 1 {
		sb.WriteString(strconv.Itoa(count))
	}
	sb.WriteRune(last)

	return sb.String()
}

func RunLengthDecode(input string) string {
	var sb strings.Builder
	sb.Grow(len(input)) // Pre-allocate memory

	count := 0
	for _, char := range input {
		if d, err := strconv.Atoi(string(char)); err == nil {
			count = count*10 + d
		} else {
			if count == 0 {
				sb.WriteRune(char)
			} else {
				sb.WriteString(strings.Repeat(string(char), count))
			}
			count = 0
		}
	}
	return sb.String()
}